syntax = "proto3";
package output;

import "google/protobuf/wrappers.proto";
message Element {
   google.protobuf.UInt64Value element = 1;
}

message Elements {
  repeated Element elements = 1; 
}

message Matrix {
  repeated Elements diagonals = 1; 
}

message Result {
  repeated Matrix matrices = 1; 
}

message Internal {
  repeated Elements simd_input = 1; 
  repeated Elements simd_output = 2; 
}

message Root {
   Result result = 1; 
   Internal internal = 2; 
}
